src/app/selection/selection.component.ts

```diff
@@ -33,18 +47,40 @@ export class SelectionComponent implements OnInit, OnDestroy {
    // this.selectItemService.deleteItem(item);
     this.currentSelectItems = this.currentSelectItems.filter(h => h !== item);
     this.items[item.id].selected = false;
+
+    if (this.currentSelectItems.length <= this.maxSelect) {
+       this.selectAbled();
+    }
   }
 
   dialogClose() {
     this.closeDialog.emit();
   }
 
+  selectDisabled() {
+    const x = document.querySelectorAll('.list-items input[type=checkbox]');
+    x.forEach(item => item.setAttribute('disabled', 'false'));
+  }
+
+  selectAbled() {
+    console.log(this.currentSelectItems.length);
+    const x = document.querySelectorAll('.list-items input[type=checkbox]');
+    x.forEach(item => item.removeAttribute('disabled'));
+  }
+
   onCheck(e) {
-    if (e.target.checked) {
-      this.currentSelectItems.push({ id: +e.target.value, title: `Item ${e.target.value}`, selected: true });
-    } else {
-      const ind = this.currentSelectItems.findIndex(value => +value.id === +e.target.value);
-      this.currentSelectItems.splice(ind, 1);
+    if (this.currentSelectItems.length <= this.maxSelect) {
+      if (e.target.checked) {
+        this.currentSelectItems.push({ id: +e.target.value, title: `Item ${e.target.value}`, selected: true });
```

Comment:
This one looks a bit messy and error-prone.
Use class constructor for the item.
Usually you would have some kind of `entites` or `infrastructure` folder in you project containing classes and interfaces specific to your domain.